src/utils/update/beneficial_owners_managing_officers_data_fetch.ts

```diff
@@ -35,23 +37,20 @@ const initialiseBoAndMoUpdateAppData = (appData: ApplicationData) => {
 
 export const retrieveBeneficialOwners = async (req: Request, appData: ApplicationData) => {
   const pscs: CompanyPersonsWithSignificantControl = await getCompanyPsc(req, appData[EntityNumberKey] as string);
-  if (pscs) {
-    for (const psc of (pscs.items || [])) {
+
+  if (pscs && pscs.items.length > 0) {
+    const boPrivateData = await getBoPrivateData(req, appData);
+    for (const psc of pscs.items) {
       logger.info("Loaded psc " + psc.kind);
-      if (psc.ceasedOn === undefined) {
-        if (psc.kind === "individual-beneficial-owner") {
-          const individualBeneficialOwner = mapPscToBeneficialOwnerTypeIndividual(psc);
-          logger.info("Loaded individual Beneficial Owner " + individualBeneficialOwner.id + " is " + individualBeneficialOwner.first_name + ", " + individualBeneficialOwner.last_name);
-          appData.update?.review_beneficial_owners_individual?.push(individualBeneficialOwner);
-        } else if (psc.kind === "corporate-entity-beneficial-owner") {
-          const beneficialOwnerOther = mapPscToBeneficialOwnerOther(psc);
-          logger.info("Loaded Beneficial Owner Other " + beneficialOwnerOther.id + " is " + beneficialOwnerOther.name);
-          appData.update?.review_beneficial_owners_corporate?.push(beneficialOwnerOther);
-        } else if (psc.kind === "legal-person-beneficial-owner") {
-          const beneficialOwnerGov = mapPscToBeneficialOwnerGov(psc);
-          logger.info("Loaded Beneficial Owner Gov " + beneficialOwnerGov.id + " is " + beneficialOwnerGov.name);
-          appData.update?.review_beneficial_owners_government_or_public_authority?.push(beneficialOwnerGov);
-        }
+      if (psc.ceasedOn !== undefined){
+        continue;
+      }
```

Comment:
The other PR  has changed to use a switch statement. 

 switch(psc.kind) {
          case "individual-beneficial-owner": 
            mapBeneficialOwnerIndividual(psc, boPrivateData, appData);
            break;
          case "corporate-entity-beneficial-owner":
             mapBeneficialOwnerOther(psc, boPrivateData, appData);
             break;
          case "legal-person-beneficial-owner":
            mapBeneficialOwnerGov(psc, boPrivateData, appData);
            break;
        }

Reply:
```switch(psc.kind) {
        case "individual-beneficial-owner":
           mapBeneficialOwnerIndividual(psc, boPrivateData, appData);
           break;
        case "corporate-entity-beneficial-owner":
          mapBeneficialOwnerOther(psc, boPrivateData, appData);
          break;
      case "legal-person-beneficial-owner":
          mapBeneficialOwnerGov(psc, boPrivateData, appData);
          break;
}
```

Reply:
thanks!